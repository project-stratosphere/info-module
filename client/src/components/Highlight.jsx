import PropTypes from 'prop-types';
import React from 'react';
import { ThumbSVG } from './SVG';
import { SVGContainer } from './styles/Summary.styles';
import {
  Wrapper,
  HelpfulContainer,
  Clickable,
  HelpfulText,
  ThanksContainer,
  HighlightText,
} from './styles/Highlight.styles';

export default class Highlight extends React.Component {
  constructor(props) {
    super(props);
    this.toggleDisplay = this.toggleDisplay.bind(this);
    this.state = {
      display: false,
    };
  }

  toggleDisplay() {
    this.setState({
      display: !this.state.display,
    });
  }

  render() {
    return (
      <Wrapper>
        <HighlightText>
          <span
            className="main"
          >
            {this.props.highlight.head}
          </span>
            &#8231;
          <span
            className="description"
          >
            {this.props.highlight.body}
          </span>
        </HighlightText>
        <span
          hidden={this.state.display}
        >
          <HelpfulContainer>
            <Clickable
              onClick={this.toggleDisplay}
              className="helpful"
            >
              <HelpfulText>
                Helpful
              </HelpfulText>
              <SVGContainer>
                <ThumbSVG />
              </SVGContainer>
            </Clickable>
            <span>
              &#8231;
            </span>
            <Clickable
              onClick={this.toggleDisplay}
              className="not-helpful"
            >
              <HelpfulText>
                Not Helpful
              </HelpfulText>
            </Clickable>
          </HelpfulContainer>
        </span>
        <span
          hidden={!this.state.display}
        >
          <ThanksContainer>
            <HelpfulText>
              Thanks for your feedback.
            </HelpfulText>
          </ThanksContainer>
        </span>
      </Wrapper>
    );
  }
}

Highlight.propTypes = {
  highlight: PropTypes.shape({
    head: PropTypes.string,
    body: PropTypes.string,
  }),
};

Highlight.defaultProps = {
  highlight: {
    head: 'String',
    body: 'String',
  },
};
