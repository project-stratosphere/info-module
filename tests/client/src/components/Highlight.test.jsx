import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Highlight from '../../../../client/src/components/Highlight';

Enzyme.configure({ adapter: new Adapter() });

const highlight = {
  head: 'Mock Highlight Head',
  body: 'Mock Highlight Body',
};

describe('Highlight', () => {
  it('should render correctly', () => {
    const output = Enzyme.shallow(<Highlight />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should change state after Helpful clicked', () => {
    const output = Enzyme.shallow(<Highlight />);
    expect(output.state().display).toEqual(false);
    output.find('.helpful').simulate('click');
    expect(output.state().display).toEqual(true);
  });
  it('should change state after Helpful clicked', () => {
    const output = Enzyme.shallow(<Highlight />);
    expect(output.state().display).toEqual(false);
    output.find('.not-helpful').simulate('click');
    expect(output.state().display).toEqual(true);
  });
});
