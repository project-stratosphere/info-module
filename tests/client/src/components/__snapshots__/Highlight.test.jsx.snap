// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight should render correctly 1`] = `
<styled.div>
  <styled.div>
    <span
      className="main"
    >
      String
    </span>
    ‧
    <span
      className="description"
    >
      String
    </span>
  </styled.div>
  <span
    hidden={false}
  >
    <styled.div>
      <styled.div
        className="helpful"
        onClick={[Function]}
      >
        <styled.div>
          Helpful
        </styled.div>
        <styled.div>
          <ThumbSVG />
        </styled.div>
      </styled.div>
      <span>
        ‧
      </span>
      <styled.div
        className="not-helpful"
        onClick={[Function]}
      >
        <styled.div>
          Not Helpful
        </styled.div>
      </styled.div>
    </styled.div>
  </span>
  <span
    hidden={true}
  >
    <styled.div>
      <styled.div>
        Thanks for your feedback.
      </styled.div>
    </styled.div>
  </span>
</styled.div>
`;
